crates/cargo-test-support/src/compare.rs

```diff
@@ -165,6 +165,11 @@ fn add_common_redactions(subs: &mut snapbox::Redactions) {
         regex::Regex::new("home/\\.cargo/registry/src/-(?<redacted>[a-z0-9]+)").unwrap(),
     )
     .unwrap();
+    subs.insert(
+        "[HASH]",
+        regex::Regex::new(r"[a-z0-9\-_]+-(?<redacted>[0-9a-f]{16})").unwrap(),
```

Comment:
This doesn't work with `-C metadata=<hash>`.

Reply:
Technically, this is trying to cover other cases so that would be another entry to cover that which wouldn't be a blocker for this PR but a nice to have

Reply:
Is there a way to forced regenerate existing snapshots, so we can add and remove redactions more freely, and won't bother forgetting to update some overly omitted snapshots?

Reply:
Not at this time.  If we remove a redaction, it'll require an update anyways.